File: java/eyes.selenium.common/src/main/java/com/applitools/eyes/locators/OcrRegion.java
package com.applitools.eyes.locators;

import com.applitools.eyes.Region;
import com.fasterxml.jackson.annotation.JsonIgnore;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

public class OcrRegion extends BaseOcrRegion {
    @JsonIgnore
    private WebElement element;
    @JsonIgnore
    private By selector;
    @JsonIgnore
    private Region region;

    public OcrRegion(WebElement element) {
        this.element = element;
    }

    public OcrRegion(By selector) {
        this.selector = selector;
    }

    public OcrRegion(Region region) {
        this.region = region;
    }

    public WebElement getElement() {
        return element;
    }

    public By getSelector() {
        return selector;
    }

    public Region getRegion() {
        return region;
    }
}

File: dotnet/Eyes.Selenium4.DotNet/Fluent/OcrRegion.cs
using Applitools.Fluent;
using Applitools.Utils.Geometry;
using OpenQA.Selenium;

namespace Applitools.Selenium
{
	public class OcrRegion : OcrRegionBase
	{
        private readonly IWebElement element_;
        private readonly By selector_;

        public OcrRegion(Region? region)
        {
            Region(region);
        }
        
        public OcrRegion(By selector)
        {
            selector_ = selector;
        }

        public OcrRegion(IWebElement element)
        {
            element_ = element;
        }

        internal By GetSelector()
        {
            return selector_;
        }

        internal IWebElement GetWebElement()
        {
            return element_;
        }
    }
}